zcash_test_vectors/transaction.py

```diff
@@ -385,6 +412,11 @@ def __init__(self, rand, consensus_branch_id):
         if have_transparent_in:
             for _ in range((rand.u8() % 3) + 1):
                 self.vin.append(TxIn(rand))
+        if is_coinbase:
+            self.vin.append(TxIn.from_components(
+                OutPoint.from_components(b'\x00' * 32, 0xFFFFFFFF),
+                Script.from_bytes(b"\x51"),
```

Comment:
To match real coinbase transactions, this should be a valid height encoding (which the current opcode is), followed by `OP_0`:
```suggestion
                Script.from_bytes(b"\x51\x00"), // [INT64(1), OP_0]
```

Reply:
The intention was to encode a height of 1, since there was no height in the genesis coinbase transaction. [According to the spec](https://zips.z.cash/protocol/protocol.pdf#txnconsensus) this should be [0x01, 0x51] (I had forgotten the length).

![image](https://user-images.githubusercontent.com/643204/151869946-0c61f44c-61dd-4d88-b06b-51eafc8cca3e.png)